cpp/src/writer/tsfile_writer.cc

```diff
@@ -393,15 +433,35 @@ int TsFileWriter::flush() {
        so map itself is ordered by device name. */
     std::map<std::string, MeasurementSchemaGroup *>::iterator device_iter;
     for (device_iter = schemas_.begin(); device_iter != schemas_.end();
-         device_iter++) {  // cppcheck-suppress postfixOperator
+         device_iter++) {
+
+        if (check_chunk_group_empty(device_iter->second)) {
+            continue;
+        }
+
         if (RET_FAIL(io_writer_->start_flush_chunk_group(device_iter->first))) {
         } else if (RET_FAIL(flush_chunk_group(device_iter->second))) {
         } else if (RET_FAIL(io_writer_->end_flush_chunk_group())) {
         }
```

Comment:
If the return value indicates an error, it should be returned immediately.

Reply:
Ok